crates/fuel-gas-price-algorithm/src/v0.rs

```diff
@@ -16,12 +16,33 @@ pub enum Error {
 pub struct AlgorithmV0 {
     /// The gas price for to cover the execution of the next block
     new_exec_price: u64,
+    /// The block height of the next L2 block
+    for_height: u32,
+    /// The change percentage per block
+    percentage: u64,
 }
 
 impl AlgorithmV0 {
     pub fn calculate(&self) -> u64 {
         self.new_exec_price
     }
+
+    #[allow(clippy::cast_possible_truncation)]
+    pub fn worst_case(&self, height: u32) -> u64 {
+        let price = self.new_exec_price as f64;
+        let blocks = height.saturating_sub(self.for_height) as f64;
+        let percentage = self.percentage as f64;
+        let percentage_as_decimal = percentage / 100.0;
+        let multiple = (1.0f64 + percentage_as_decimal).powf(blocks);
+        let mut approx = price * multiple;
```

Comment:
Are we sure that this function will not overflow during any of the operations?

Reply:
The prop test should ensure that it will never happen.

Reply:
But do we limit the block horizon that comes from the user? What if it is bigger than 10_000?
<img width="690" alt="image" src="https://github.com/user-attachments/assets/8c067192-e1ec-4b53-b263-545d1c7a0fb6">

Reply:
You're right. We don't check that (because the test iterates over that entire size and I don't want to do `u32` max in a test).

I can add another test that makes sure it never overflows :)